### PR TITLE
template: cypress test

### PR DESCRIPTION
**Pull request by zannkukai**
Implements Cypress to about template create and utilisation.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

File: tests/e2e/cypress/cypress/integration/templates/templates-test.spec.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  after('Logout user', function() {
+    cy.api_logout()
+  })
+
+  it('Create a new template and use it in the editor', function() {
+    const template = this.templates.templateA
+
+    cy.visit('/professional/records/documents/new')
+    cy.wait(2000)
```
Comment (AoNoOokami) [inline tests/e2e/cypress/cypress/integration/templates/templates-test.spec.js:59]: Is it necessary to use wait() ? Normally there is a 2 min timeout for cy.get() command.

File: tests/e2e/cypress/cypress/integration/templates/templates-test.spec.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    cy.get('#editor-save-button-dropdown-split')
+      .find('li a.dropdown-item:nth-child(1)')  // TODO: Find a better way to retrieve the correct link to click
+      .click()
+    cy.get('.modal-content #name').type(template.name)
+    cy.get('.modal-content button:submit').click()
+    cy.wait('@api_template_create')
+    cy.url().should('include', 'records/templates/detail')
+
+    cy.visit('/professional/records/documents/new')
+    cy.wait(2000)
```
Comment (AoNoOokami) [inline tests/e2e/cypress/cypress/integration/templates/templates-test.spec.js:72]: Same comment here

File: tests/e2e/cypress/cypress/support/api.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  *  :param user - object: a user object from the 'users' fixtures file
+  *  :param password - string: the password to log in
+  */
+Cypress.Commands.add('api_login', (user, password) => {
+  cy.request('/login')
+  .its('body')
+  .then(body => {
+    const $html = Cypress.$(body)
+    const csrf_token = $html.find('input[name=csrf_token]').val()
+    cy.log(csrf_token)
```
Comment (AoNoOokami) [inline tests/e2e/cypress/cypress/support/api.js:39]: I think it's a good thing to use api calls instead of UI actions. I also note the cy.log() command :)

**Review by AoNoOokami — CHANGES_REQUESTED** [review]

File: tests/e2e/cypress/cypress/integration/templates/templates-test.spec.js (hunk 1 above)
Comment (zannkukai) [reply tests/e2e/cypress/cypress/integration/templates/templates-test.spec.js:59]: @jma told me some weeks ago than we can't know when the build of the editor has really finished due to a lot of asynchronous calls. So the `visit` return quickly a HTTP 200 response but the editor is still in construction, adding a `wait` for 2 seconds allows to give me more chance than the form building is completed.

File: tests/e2e/cypress/cypress/support/api.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+*/
+
+/** API call to logout method ============================================== */
+Cypress.Commands.add('api_logout', () => {
```
Comment (AoNoOokami) [inline tests/e2e/cypress/cypress/support/api.js:21]: Can we keep a camelCase for commands designation as done in other commands? -> `apiLogout`

File: tests/e2e/cypress/cypress/support/api.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  })
+})
+
+
+/** API call to delete items ==================================================
+  * Delete resource items based on a query
+  * :param resourceName - string: the resource type
+  * :param query - string: criteria to find items to delete
+  */
+Cypress.Commands.add('api_delete_items', (resourceName, query) => {
```
Comment (AoNoOokami) [inline tests/e2e/cypress/cypress/support/api.js:63]: `apiDeleteItems` ?

File: tests/e2e/cypress/cypress/support/api.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    url: '/logout/',
+    followRedirect: false
+  })
+})
+
+/** API call to login method ==================================================
+  *  :param user - object: a user object from the 'users' fixtures file
+  *  :param password - string: the password to log in
+  */
+Cypress.Commands.add('api_login', (user, password) => {
```
Comment (AoNoOokami) [inline tests/e2e/cypress/cypress/support/api.js:33]: `apiLogin`

**Review by AoNoOokami — CHANGES_REQUESTED** [review]

File: tests/e2e/cypress/cypress/integration/templates/templates-test.spec.js (hunk 1 above)
Comment (AoNoOokami) [reply tests/e2e/cypress/cypress/integration/templates/templates-test.spec.js:59]: Ok

File: tests/e2e/cypress/cypress/support/api.js (hunk 5 above)
Comment (zannkukai) [reply tests/e2e/cypress/cypress/support/api.js]: I used `apiDeleteResources` to be more generic

**Review by AoNoOokami — APPROVED** [review]

**Comment by AoNoOokami** [comment]
Tested locally and approved (some other tests were broken with recent merge of US but a fix is proposed here: https://github.com/rero/rero-ils/pull/1260)

**Review by benerken — APPROVED** [review]

**Pull request merged**